spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -138,6 +138,7 @@ public class TestRewriteDataFilesAction extends TestBase {
   public static void setupSpark() {
     // disable AQE as tests assume that writes generate a particular number of files
     spark.conf().set(SQLConf.ADAPTIVE_EXECUTION_ENABLED().key(), "false");
+    spark.conf().set(SQLConf.ANSI_ENABLED().key(), "false");
```

Comment:
Why is this needed?

Reply:
I actually meant to change only the ANSI setting for `TestRewriteDataFilesAction#testZOrderAllTypesSort`, because CAST(longCol AS BINARY) is not allowed with ANSI, which is the default now for Spark 4.0. 
I fixed the test to use ANSI  false for testZOrderAllTypesSort.